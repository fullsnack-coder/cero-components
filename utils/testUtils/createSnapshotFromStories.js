import React from 'react'
import { render } from '@testing-library/react'
import { composeStories } from '@storybook/testing-react'

// TODO: Move this to a generic approach where it
// reads all the stories and works as Storyshoot
const createSnapshotFromStories = (componentName, stories) => {
  const composedStories = composeStories(stories)

  Object.entries(composedStories).forEach(([story, Component]) => {
    describe(`when '${componentName}' ${story} is mounted`, () => {
      it('should render', () => {
        const { asFragment } = render(<Component />)
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
}

export default createSnapshotFromStories
