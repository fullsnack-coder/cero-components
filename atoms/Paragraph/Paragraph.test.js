import * as stories from './Paragraph.stories'
import createSnapshotFromStories from '../../utils/testUtils/createSnapshotFromStories'

describe('[ atoms / Paragraph ]', () => {
  describe('#Paragraph', () => createSnapshotFromStories('Paragraph', stories))
})
