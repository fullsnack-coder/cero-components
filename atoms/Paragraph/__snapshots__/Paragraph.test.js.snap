// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ atoms / Paragraph ] #Paragraph when 'Paragraph' Centered is mounted should render 1`] = `
<DocumentFragment>
  <p
    class="paragraph color-base size-md weight-normal is-centered"
  >
    Lose consequence night studies Wolves should wizards destruction burglar? All right, then. Keep your secrets.
  </p>
</DocumentFragment>
`;

exports[`[ atoms / Paragraph ] #Paragraph when 'Paragraph' Colors is mounted should render 1`] = `
<DocumentFragment>
  <p
    class="paragraph color-base size-md weight-normal"
  >
    Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange Udùn piled fighters potent?
  </p>
  <p
    class="paragraph color-muted size-md weight-normal"
  >
    Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange Udùn piled fighters potent?
  </p>
  <p
    class="paragraph color-inverted size-md weight-normal"
  >
    Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange Udùn piled fighters potent?
  </p>
  <p
    class="paragraph color-primary size-md weight-normal"
  >
    Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange Udùn piled fighters potent?
  </p>
  <p
    class="paragraph color-tertiary size-md weight-normal"
  >
    Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange Udùn piled fighters potent?
  </p>
</DocumentFragment>
`;

exports[`[ atoms / Paragraph ] #Paragraph when 'Paragraph' Default is mounted should render 1`] = `
<DocumentFragment>
  <p
    class="paragraph color-base size-md weight-normal"
  >
    Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange Udùn piled fighters potent?
  </p>
</DocumentFragment>
`;

exports[`[ atoms / Paragraph ] #Paragraph when 'Paragraph' Inline is mounted should render 1`] = `
<DocumentFragment>
  <p
    class="paragraph color-base size-md weight-normal is-inline"
  >
    Swords are no more.
  </p>
</DocumentFragment>
`;

exports[`[ atoms / Paragraph ] #Paragraph when 'Paragraph' Monospace is mounted should render 1`] = `
<DocumentFragment>
  <p
    class="paragraph color-base size-md weight-normal is-monospace"
  >
    Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange Udùn piled fighters potent?
  </p>
</DocumentFragment>
`;

exports[`[ atoms / Paragraph ] #Paragraph when 'Paragraph' Sizes is mounted should render 1`] = `
<DocumentFragment>
  <p
    class="paragraph color-base size-xs weight-normal"
  >
    Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange Udùn piled fighters potent?
  </p>
  <p
    class="paragraph color-base size-sm weight-normal"
  >
    Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange Udùn piled fighters potent?
  </p>
  <p
    class="paragraph color-base size-md weight-normal"
  >
    Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange Udùn piled fighters potent?
  </p>
  <p
    class="paragraph color-base size-lg weight-normal"
  >
    Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange Udùn piled fighters potent?
  </p>
</DocumentFragment>
`;

exports[`[ atoms / Paragraph ] #Paragraph when 'Paragraph' Striked is mounted should render 1`] = `
<DocumentFragment>
  <p
    class="paragraph color-base size-md weight-normal is-striked"
  >
    Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange Udùn piled fighters potent?
  </p>
</DocumentFragment>
`;

exports[`[ atoms / Paragraph ] #Paragraph when 'Paragraph' Weights is mounted should render 1`] = `
<DocumentFragment>
  <p
    class="paragraph color-base size-md weight-normal"
  >
    Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange Udùn piled fighters potent?
  </p>
  <p
    class="paragraph color-base size-md weight-medium"
  >
    Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange Udùn piled fighters potent?
  </p>
  <p
    class="paragraph color-base size-md weight-semibold"
  >
    Swords are no more use here. Esquire days mountain Fangorn champion brace waiting laboring preparing! Canopy plunder deceit cakehole strange Udùn piled fighters potent?
  </p>
</DocumentFragment>
`;
